packages/react-dom/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -1189,6 +1236,39 @@ function pushStartTitle(
   return children;
 }
 
+function pushStartHead(
+  target: Array<Chunk | PrecomputedChunk>,
+  preamble: ?Array<Chunk | PrecomputedChunk>,
+  props: Object,
+  tag: string,
+  responseState: ResponseState,
+): ReactNodeList {
+  // Preamble type is nullable for feature off cases but is guaranteed when feature is on
```

Comment:
I usually just make the type contain everything you'd expect to exist in the new common state.

Reply:
I was trying to avoid two extra array allocations for non dom (even though no one else is really using it yet I suppose). In the future I'd expect that preamble is not nullable for dom fizz server and is explicitly null for environments that do not support preamble but how would I type that? I guess I could export the preamble type but then I'd  probably also need a create function for the preamble itself.

Do you recommend doing it that way?

Reply:
It'd just make it required. It's at most two per request, and we don't know if other renderers will end up using it yet.